### PR TITLE
Add phone number formatting

### LINKED ISSUE #439: Phone number formatting in demographics eg (123)456-7890.
Phone number formatting in demographics example (123)456-7890. This should be user definable using a global variable.

### PR DESCRIPTION
**Pull request by azwreith**
Fixes #439 

A limitation of this code is that it assumes all formatting options to be added in the future to be in `3 3 4` , i.e, US format. An advantage of this is that people can add new formats by just including it in the `globals.inc.php` file. If need be, I can expand this code for the general case, but that model would require extra effort for people using the system to add new formats (since they'll have to write regex). Opinions?

The check for phone numbers is done (for now) using layout_options.edit_options value

EDIT: Please check lines 829-842 of global_inc.php and lines 1705-1715 of options_inc.php for the diffs

### CODE REVIEW

**Comment by teryhill** [comment]
@azwreith can you put a check in to see if the country code for the phone is set to US (Telephone Country Code in globals) so it will still work for everyone else

**Comment by azwreith** [comment]
@teryhill It does work for other countries as well. The thing is that only 3 3 4 type formats can be made using this. Other countries might have formats like XXX XXXX XXX which is a 3 4 3 format and can't be applied using this code. I can put the country code check in but then it'll ignore the setting although for other countries. If that's fine, I'll put it in c:

**Comment by teryhill** [comment]
I am sorry I misunderstood, as long as it still works for other countries. The country code check is not needed then.

**Comment by azwreith** [comment]
Haha sorry I should have been clear about what I meant in the first place >_<

**Comment by teryhill** [comment]
Not your problem I am just way old 8^)

**Comment by aethelwulffe** [comment]
Let us know when you are done so that we can go over, test and approve these changes.

File: library/globals.inc.php [hunk 1]
```diff
@@ -826,9 +826,22 @@
...
+      array(
+        '($1) $2-$3' => '(XXX) XXX-XXXX',   // $1: 3 digits, $2: 3 digits, $3: 4 digits
+        '($1) $2 $3' => '(XXX) XXX XXXX',
+        '$1-$2-$3' => 'XXX-XXX-XXXX',
+        '$1 $2 $3' => 'XXX XXX XXXX',
+        '$1$2$3' => 'XXXXXXXXXX'
+      ),
+      '($1) $2-$3',                    //default
+      xl('The format to display phone numbers in.')
+    )
```
Comment (teryhill) [inline library/globals.inc.php]: this needs to be ),

**Comment by teryhill** [comment]
@azwreith  I added your code , made sure there was a "P" in the layouts. The phone number will not show on the main screen, You can enter it and it stays but is not displayed.

image

File: library/globals.inc.php (hunk 1 above)
Comment (azwreith) [inline library/globals.inc.php]: Um it's the last one in the list so nuh :sweat_smile:

**Comment by azwreith** [comment]
@teryhill You have to go once in the Globals settings menu, set a phone number format, and save. As the default value for phone number isn't in the database. I encountered the same issue while coding it :smiley:

**Comment by teryhill** [comment]
Yep I forgot about that. It is going to be a big pain when clients start upgrading, We are discussing a way to do the upgrades that is easy to code and maintain for us and include a rollback mechanism for the clients in case there is a problem. Good work it test well. Can you come up with a way of using it for the other phone numbers in the system like Insurance, facility, address book , users ( i don't remember if that's all or not)

**Comment by azwreith** [comment]
One more thing, when you click on edit on demographics page, you go to the page where you can change demographics settings. There, the phone numbers are still there in the xxx-xxx-xxxx format. I did that intentionally because it'll be easier for the person to edit their number. Here's a SS for the same:
screenshot from 2017-03-03 07-53-52
vs
screenshot from 2017-03-03 07-56-48

**Comment by teryhill** [comment]
@tmccormi  @aethelwulffe I up vote this

**Comment by azwreith** [comment]
Well, all I can think right now is to manually code it in for each of them. I'll brainstorm more after my morning lab! :sob:

**Comment by azwreith** [comment]
Well, we could have a common global formatting function (maybe in `formatting.inc.php`) and call upon that wherever needed. And we could control that using a common global variable.

**Comment by teryhill** [comment]
@azwreith  can you check the S.S.N field and see why it is updating with the phone number formatting?

**Comment by azwreith** [comment]
@teryhill Ohh I just noticed that :O Lemme see why..

**Comment by teryhill** [comment]
The mobile phone is not taking the change also.

**Comment by azwreith** [comment]
@teryhill It is, phone numbers needs to have the standard 10 digits or else they format can't be applied to them.

**Comment by teryhill** [comment]
You are correct, I had only 9 digits, some of that old guy stuff

**Comment by azwreith** [comment]
@teryhill Hehe, I'm sure you're quite young and energetic :smile:

File: library/globals.inc.php (hunk 1 above)
Comment (aethelwulffe) [inline library/globals.inc.php]: He got you there...Line 830.

**Comment by aethelwulffe** [comment]
This stomps on earlier unmerged commits, and they will have to be cleaned up (again), but this is simple, clean, and on the issues list.  Merging.  Shoot me later if you like. :)

**Pull request merged**